From 10cf654df2797dcab82efb4a9f62069621f42c9f Mon Sep 17 00:00:00 2001
From: lghchinaidea <lghchinaidea@gmail.com>
Date: Tue, 4 Jun 2024 11:06:38 +0800
Subject: [PATCH] android build: need to remove libpthread

---
 src/kj/BUILD.bazel | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/kj/BUILD.bazel b/src/kj/BUILD.bazel
index de89d41e..5b6da397 100644
--- a/src/kj/BUILD.bazel
+++ b/src/kj/BUILD.bazel
@@ -74,10 +74,11 @@ cc_library(
     linkopts = select({
         "@platforms//os:windows": [],
         ":use_libdl": [
-            "-lpthread",
+            #"-lpthread",
             "-ldl",
         ],
-        "//conditions:default": ["-lpthread"],
+        #"//conditions:default": ["-lpthread"],
+		"//conditions:default": [],
     }),
     visibility = ["//visibility:public"],
     deps = [":kj-defines"],
-- 
2.36.0.windows.1

